semantic_model_generator/data_processing/cte_utils.py

```diff
@@ -112,8 +122,57 @@ def _get_col_expr(column: semantic_model_pb2.Column) -> str:
     return cte
 
 
+def _convert_to_snowflake_sql(sql: str) -> str:
+    """
+    Converts a given SQL statement to Snowflake SQL syntax using SQLGlot.
+
+    Args:
+    sql (str): The SQL statement to convert.
+
+    Returns:
+    str: The SQL statement in Snowflake syntax.
+    """
+    try:
+        expression = sqlglot.parse_one(sql, dialect=Snowflake)
+    except Exception as e:
+        raise ValueError(
+            f"Unable to parse sql statement.\n Provided sql: {sql}\n. Error: {e}"
+        )
+
+    return expression.sql()
+
+
+def generate_select(
+    table_in_column_format: semantic_model_pb2.Table, limit: int
+) -> str:
+    """Generate select query for all columns except for columns with aggregations, for validation purpose."""
+    cte = _generate_cte_for(table_in_column_format)
+    sql = (
+        cte
+        + f"SELECT * FROM {logical_table_name(table_in_column_format)} LIMIT {limit}"
+    )
+    return _convert_to_snowflake_sql(sql)
+
+
+def generate_agg_expr_selects(table: semantic_model_pb2.Table, limit: int) -> List[str]:
```

Comment:
instead of creating new sql for every expr, can you add the _get_col_expr to the original sql so we can just run one sql statement

Reply:
i am not sure i follow - each expr have to be verified separately if having an aggregation

Reply:
if no aggregation, it's one sql in generate_select

Reply:
I'm also not following why an aggregation expression needs to be checked separately. Can you add the explanation to the docstring?

Reply:
oh actually if all aggregate without dimensions it can be one. Just cannot mix non-aggregate vs. aggregate. Let me fix

Reply:
@sfc-gh-jhilgart @sfc-gh-nlimtiaco updated, PTAL
Basically have one sql that generates select on all column expr without aggregation; and another sql on all column exprs with aggregation